src/main/java/com/nokia/uwr/service/turns/TurnHandlerImpl.java

```diff
@@ -0,0 +1,81 @@
+package com.nokia.uwr.service.turns;
+
+import com.nokia.uwr.board.Board;
+import com.nokia.uwr.model.UE;
+import com.nokia.uwr.scenario.ScenarioSchema;
+import com.nokia.uwr.scenario.ue.UEAction;
+import com.nokia.uwr.scenario.ue.UEScenario;
+import com.nokia.uwr.scenario.ue.UEStep;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Service;
+
+/**
+ * Class that performs actions for certain turns.
+ *
+ * @author najedzony
+ */
+@Service
+public class TurnHandlerImpl implements TurnHandler {
+
+    private static final Logger LOGGER = LogManager.getLogger(Board.class);
+
+    private final Board board;
+    private ScenarioSchema scenario;
+
+    public TurnHandlerImpl(Board board) {
+        this.board = board;
+    }
+
+    /**
+     * Function that sends a signal to CALLSYS about start of a new call.
+     *
+     * @author najedzony
+     */
+    private void StartCall() {
+        /* TODO */
+    }
+
+    /**
+     * Function that sends a signal to CALLSYS about end of an ongoing call.
+     *
+     * @author najedzony
+     */
+    private void EndCall() {
+        /* TODO */
+    }
+
+    @Override
+    public void FindAndDoActionsForThisTurn(int TurnNumber) {
+        for (UEScenario ueScenario : scenario.ueScenarios()) {
+            for (UEStep ueStep : ueScenario.steps()) {
```

Comment:
I think it's better solution, to check only first step of ueScenario.steps(), because others will take place for sure in other turns. Moreover, I would delete completed step from this list, so we don't need to check them later again. From this it follows that the steps list can be created as LinkedList or Queue.